tools/src/main/java/org/apache/kafka/trogdor/coordinator/NodeManager.java

```diff
@@ -94,9 +97,15 @@
             this.state = state;
         }
 
+        boolean hasExpired() {
+            return spec.hasExpired(time, startedMs);
+        }
+
         void tryCreate() {
             try {
                 client.createWorker(new CreateWorkerRequest(workerId, taskId, spec));
+                if (startedMs == -1)
+                    startedMs = time.milliseconds();
```

Comment:
Here the Coordinator keeps track of when it has first ran the task - useful in cases where `spec.startMs==0`